src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/OpportunityNotificationMetadataParser.java

```diff
@@ -75,4 +80,138 @@ protected void postProcessOutputData(EntityObject output, JsonNode input) {
+    }
+  }
+
+  private void proccessOwnerNameAndEmailFormatted(JsonNode node) {
+    String ownerNameAndEmailFormatted = EMPTY;
+
+    String ownerName = node.path(SalesforceConstants.NAME).asText(EMPTY);
+    if (!StringUtils.isEmpty(ownerName)) {
+      ownerNameAndEmailFormatted = ownerName;
+    }
+
+    String ownerEmail = node.path(SalesforceConstants.EMAIL).asText(EMPTY);
+    if (!StringUtils.isEmpty(ownerEmail)) {
+      if (!StringUtils.isEmpty(ownerNameAndEmailFormatted)) {
+        ownerNameAndEmailFormatted = ownerNameAndEmailFormatted + SEPARATOR + ownerEmail;
+      } else {
+        ownerNameAndEmailFormatted = ownerEmail;
+      }
+    }
+
+   ((ObjectNode) node).put(SalesforceConstants.NAME_AND_EMAIL, ownerNameAndEmailFormatted);
+  }
+
+  private String getAmountFormatted(JsonNode node) {
+    String amount = node.path(SalesforceConstants.AMOUNT).asText(EMPTY);
+
+    if (!StringUtils.isEmpty(amount)) {
+      amount = NumberFormatUtils.formatValueWithLocale(Locale.US, amount);
+    }
+
+    return amount;
+  }
+
+  private void proccessAmountAndCurrencyIsoCode(JsonNode node) {
+    String amountAndCurrencyIsoCodeFormatted = EMPTY;
+
+    String amount = getAmountFormatted(node);
+    if (!StringUtils.isEmpty(amount)) {
+      amountAndCurrencyIsoCodeFormatted = amount;
+    }
+
+    String currencyIsoCode = node.path(SalesforceConstants.CURRENCY_ISO_CODE).asText(EMPTY);
+    if (!StringUtils.isEmpty(currencyIsoCode)) {
+      if (!StringUtils.isEmpty(amountAndCurrencyIsoCodeFormatted)) {
+        amountAndCurrencyIsoCodeFormatted = amountAndCurrencyIsoCodeFormatted + SEPARATOR + currencyIsoCode;
+      } else {
+        amountAndCurrencyIsoCodeFormatted = currencyIsoCode;
+      }
+    }
+
+    ((ObjectNode) node).put(SalesforceConstants.AMOUNT_AND_CURRENCY_ISO_CODE, amountAndCurrencyIsoCodeFormatted);
+  }
+
+  private void processAccountName(JsonNode node) {
+    String accountName = node.path(SalesforceConstants.NAME).asText(EMPTY);
+
+    if (StringUtils.isEmpty(accountName)) {
+      ((ObjectNode) node).put(SalesforceConstants.NAME, DEFAULT_VALUE_NULL);
+    }
+  }
+
+  private boolean emailExistsAtSymphony(String emailAddress) {
+    if (StringUtils.isBlank(emailAddress)) {
+      return false;
+    }
+
+    User user = userService.getUserByEmail(integrationUser, emailAddress);
+    return user.getId() != null;
+  }
+
+  private void proccessCloseDate(JsonNode node) {
+    String closeDateFormat = node.path(SalesforceConstants.CLOSE_DATE).asText(null);
+    SimpleDateFormat formatter = new SimpleDateFormat(SalesforceConstants.TIMESTAMP_FORMAT);
+
+    if (!StringUtils.isEmpty(closeDateFormat)) {
+      try {
+        closeDateFormat = formatter.format(formatter.parse(closeDateFormat));
+      } catch (ParseException e) {
+        // Do nothing
+      }
+
+      ((ObjectNode) node).put(SalesforceConstants.CLOSE_DATE, closeDateFormat);
+    }
+  }
+
+  private void proccessURLIconIntegration(JsonNode node) {
```

Comment:
You could concentrate the whole logic for URLIcon in the super class, even the JSON manipulation, since every subclass must have the same JSON property as output.